File: 기초-브론즈 문제/백준_2798_블랙잭/블랙잭.kt
import java.util.*

fun main(args: Array<String>) {
    val s = Scanner(System.`in`)
    val n = s.nextInt()
    val m = s.nextInt()
    var arr = IntArray(100, { 0 })

    for(i in 0..n-1) arr[i] = s.nextInt()

    var max = 0;
    for(i in 0..n-3){
        for(j in i+1..n-2){
            for(k in j+1.. n-1){
                var sum = arr[i] + arr[j] + arr[k]
                if(sum <= m){
                    if(max < sum) max = sum
                }
            }
        }
    }
    print(max)

}

File: 기초-브론즈 문제/백준_2798_블랙잭/블랙잭.cs
using System;
using static System.Console;

class Program
{
    public static void Main(string[] args)
    {
        int n, m;
        int[] arr = new int[100];
        string[] s = ReadLine().Split(' ');
        n = int.Parse(s[0]);
        m = int.Parse(s[1]);

        string[] nums = ReadLine().Split(' ');
        for (int i = 0; i < n; i++) arr[i] = int.Parse(nums[i]);

            int max = 0;
        for(int i = 0; i < n-2; i++)
        {
            for (int j = i + 1; j < n - 1; j++)
            {
                for (int k = j + 1; k < n; k++)
                {
                    int sum = arr[i] + arr[j] + arr[k];
                    if (sum <= m)
                    {
                        if (max < sum) max = sum;
                    }
                }
            }
        }

        WriteLine(max);
    }
}
